benchmarking/run.py

```diff
@@ -0,0 +1,263 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+from pathlib import Path
+import yaml
+import os
+import time
+import sys
+from typing import Any
+import json
+import traceback
+from statistics import mean, stdev
+import pickle
+import shutil
+import traceback
+
+from loguru import logger
+
+from nemo_curator.tasks import Task
+from nemo_curator.tasks.utils import TaskPerfUtils
+
+# FIXME: How do we want to package this tool? Perhaps a package extra for
+#  nemo-curator, i.e. nemo-curator[benchmarking]?
+# For now, add this directory to PYTHONPATH to import the runner modules
+sys.path.insert(0, Path(__file__).parent)
+from runner.matrix import MatrixConfig, MatrixEntry    
+from runner.datasets import DatasetResolver
+from runner.utils import get_obj_for_json
+from runner.process import run_command_with_timeout
+from runner.env_capture import dump_env
+
+
+def ensure_dir(dir_path: Path) -> None:
+    """Ensure dir_path and parents exists, creating them if necessary."""
+    dir_path.mkdir(parents=True, exist_ok=True)
+
+
+def create_or_overwrite_dir(dir_path: Path) -> None:
+    """Create directory, removing it if it exists."""
+    if dir_path.exists():
+        shutil.rmtree(dir_path, ignore_errors=True)
+    dir_path.mkdir(parents=True, exist_ok=True)
```

Comment:
you can change this PR or add a todo, but this can just be `create_or_overwrite_dir` from `nemo_curator.utils.file_utils`